packages/core/security/core-security-server-internal/src/fips/fips.ts

```diff
@@ -9,28 +9,70 @@
 
 import type { Logger } from '@kbn/logging';
 import { getFips } from 'crypto';
-import { SecurityServiceConfigType } from '../utils';
+import { PKCS12ConfigType, SecurityServiceConfigType } from '../utils';
 
 export function isFipsEnabled(config: SecurityServiceConfigType): boolean {
   return config?.experimental?.fipsMode?.enabled ?? false;
 }
 
-export function checkFipsConfig(config: SecurityServiceConfigType, logger: Logger) {
+export function checkFipsConfig(
+  config: SecurityServiceConfigType,
+  elasticsearchConfig: PKCS12ConfigType,
+  serverConfig: PKCS12ConfigType,
+  logger: Logger
+) {
   const isFipsConfigEnabled = isFipsEnabled(config);
   const isNodeRunningWithFipsEnabled = getFips() === 1;
 
   // Check if FIPS is enabled in either setting
   if (isFipsConfigEnabled || isNodeRunningWithFipsEnabled) {
-    // FIPS must be enabled on both or log and error an exit Kibana
+    const definedPKCS12ConfigOptions = findDefinedPKCS12ConfigOptions(
+      elasticsearchConfig,
+      serverConfig
+    );
+    // FIPS must be enabled on both, or, log/error an exit Kibana
     if (isFipsConfigEnabled !== isNodeRunningWithFipsEnabled) {
       logger.error(
         `Configuration mismatch error. xpack.security.experimental.fipsMode.enabled is set to ${isFipsConfigEnabled} and the configured Node.js environment has FIPS ${
           isNodeRunningWithFipsEnabled ? 'enabled' : 'disabled'
         }`
       );
+
+      process.exit(78);
```

Comment:
Nit: I know `process.exit` is pre-existing, but I thought we could  [leverage the "shutdown" listener](https://github.com/elastic/kibana/blob/462e4378100e8d5233e7379f9244da5609b7dd2c/packages/core/root/core-root-server-internal/src/bootstrap.ts#L132) by throwing a `CriticalError` like [this](https://github.com/elastic/kibana/blob/462e4378100e8d5233e7379f9244da5609b7dd2c/packages/core/root/core-root-server-internal/src/server.ts#L417) that allows providing a message and an exit code.

Happy to stick with the current approach, my primary thought was just to give Core the chance to run the shutdown lifecycle and might make testing a little bit easier 🤷🏻

Reply:
I like the `CriticalError` approach, thanks for showing us this, @jloleysens!